src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -292,5 +309,127 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
+                // file(s), the process could be gone.
+                result = default(ParsedStatus);
+                return false;
+            }
+
+            var results = default(ParsedStatus);
+            while (!statusFileContents.IsEmpty)
+            {
+                int startIndex = statusFileContents.IndexOf(':');
+                int endIndex = statusFileContents.IndexOf('\n');
+
+                // Reached end of file
+                if (endIndex == -1)
+                {
+                    break;
+                }
+
+                ReadOnlySpan<char> title = statusFileContents.Slice(0, startIndex);
+                ReadOnlySpan<char> value = default;
+                // endIndex - startIndex - 1 --> To slice till just before '\n'
+                // endIndex - startIndex - 4 --> To slice till just before ' kB\n'
+                switch (title.ToString())
+                {
+#if DEBUG
+                    case "Pid":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 1);
+                        int.TryParse(value, out results.Pid);
+                        break;
+#endif
+                    case "VmHWM":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 4);
+                        ulong.TryParse(value, out results.VmHWM);
+                        break;
+
+                    case "VmRSS":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 4);
+                        ulong.TryParse(value, out results.VmRSS);
+                        break;
+
+                    case "VmData":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 4);
+                        ulong.TryParse(value, out results.VmData);
+                        break;
+
+                    case "VmSwap":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 4);
+                        ulong.TryParse(value, out results.VmSwap);
+                        break;
+
+                    case "VmSize":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 4);
+                        ulong.TryParse(value, out results.VmSize);
+                        break;
+
+                    case "VmPeak":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 4);
+                        ulong.TryParse(value, out results.VmPeak);
+                        break;
+
+                    case "VmStk":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 4);
+                        ulong.TryParse(value, out ulong VmStk);
+                        // Add stack value to VmData to calculate memory usage
+                        results.VmData += VmStk;
```

Comment:
What happens if "VmData" is parsed after this... won't it overwrite what was written here?

Reply:
Although, the linux source code which has not changed from decades guarantees the order in which we get values, I agree that adding values safely would be better. Will  update this.

Reply:
Updated `VmData` case to add to the value as well instead of assigning. Adding won't hurt as default value here is `0`.